spec/minefield_spec.rb

```diff
@@ -0,0 +1,282 @@
+        expect(t.field[2][0].adjacent_mines).to be 0
+        expect(t.field[2][1].adjacent_mines).to be 0
+        expect(t.field[2][2].adjacent_mines).to be 0
+      end
+      it 'correctly labels neighbors of two mines' do
+        t.field[0][0].place_mine
+        t.field[0][2].place_mine
+        t.generate_adjacent_mine_counts
+        expect(t.field[0][0].adjacent_mines).to be 0
+        expect(t.field[0][1].adjacent_mines).to be 2
+        expect(t.field[0][2].adjacent_mines).to be 0
+        expect(t.field[1][0].adjacent_mines).to be 1
+        expect(t.field[1][1].adjacent_mines).to be 2
+        expect(t.field[1][2].adjacent_mines).to be 1
+        expect(t.field[2][0].adjacent_mines).to be 0
+        expect(t.field[2][1].adjacent_mines).to be 0
+        expect(t.field[2][2].adjacent_mines).to be 0
+      end
+      it 'correctly labels neighbors of multiple mines' do
+        t.field[0][0].place_mine
+        t.field[0][2].place_mine
+        t.field[1][0].place_mine
+        t.field[2][0].place_mine
+        t.generate_adjacent_mine_counts
+        expect(t.field[0][0].adjacent_mines).to be 1
+        expect(t.field[0][1].adjacent_mines).to be 3
+        expect(t.field[0][2].adjacent_mines).to be 0
+        expect(t.field[1][0].adjacent_mines).to be 2
+        expect(t.field[1][1].adjacent_mines).to be 4
+        expect(t.field[1][2].adjacent_mines).to be 1
+        expect(t.field[2][0].adjacent_mines).to be 1
+        expect(t.field[2][1].adjacent_mines).to be 2
+        expect(t.field[2][2].adjacent_mines).to be 0
+      end
+    end
+  end
+
+  describe '#field' do
+    it 'is an array' do
+      expect(m.field).to be_a_kind_of Array
+    end
+    it 'has as many rows as the field size' do
+      expect(m.field.length).to eq m.size
+    end
+    it 'each row is as long as the field size' do
+      expect(m.field.all? { |row| row.length == m.size }).to be true
+    end
+    it 'has a cell object in every cell' do
+      expect(m.field.flatten.all? { |cell| cell.class == Cell }).to be true
+    end
+    it 'cells have unique object IDs' do
+      expect(m.field[0][0]).not_to equal m.field[5][5]
+      expect(m.field[0][9]).not_to equal m.field[9][0]
+      expect(m.field[3][7]).not_to equal m.field[7][3]
+    end
+  end
+
+  describe 'private methods' do
+    it '#generate_mines should be private' do
+      expect do
+        m.generate_mines
+      end.to raise_error
+    end
+    it '#generate_adjacent_mine_counts should be private' do
+      expect do
+        m.generate_adjacent_mine_counts
+      end.to raise_error
+    end
+  end
+
+  describe 'take_turn' do
+    let(:toy){ ToyMinefield.new }
+
+    before(:each) do
+      toy.field[0][0].place_mine
+      toy.generate_adjacent_mine_counts
+    end
+
+    context '#clear' do
+      it 'clears a cell with an adjacent mine' do
+        toy.take_turn({row: 0, column: 1, action: 'C'})
+        expect(toy.field[0][1].cleared).to be true
+      end
```

Comment:
It took me a while to understand clearly that "clear" means the player making a non-flag move on the board.... I guess it makes sense though because that is technically what the player is doing.